javascript/organizing_your_javascript_code/webpack.md

```diff
@@ -1,24 +1,62 @@
+
+An example of how to set the template option within your webpack configuration file is as follows:
+
+~~~javascript
+// webpack.config.js
+
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+// ...
+    plugins: [
+        new HtmlWebpackPlugin({
+            template: './src/template.html',
+        }),
+    ],
+// ...
+~~~
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
 1. Code along with [this tutorial](https://webpack.js.org/guides/asset-management/) to see examples of using webpack to manage your website's assets.
-2. Code along with [this tutorial](https://webpack.js.org/guides/output-management/) to learn how to let webpack automatically manage your index.html and insert your bundle into the page for you!
-3. Finally, the first part of [this tutorial](https://webpack.js.org/guides/development/) talks about source maps, a handy way to track down which source file (index.js, a.js, b.js) an error is coming from when you use webpack to bundle them together. This is essential to debugging bundled code in your browser's DevTools. If the error comes from b.js the error will reference that file instead of the bundle. It also walks through an example of the `--watch` feature you _definitely_ should have taken note of above.
-4. You don't need to do the rest of the webpack tutorials at this time, but take a look at what's offered on the sidebar of their [guides page](https://webpack.js.org/guides/). There are several sweet features that you might want to use in future projects such as code-splitting, lazy-loading, and tree-shaking. Now that you have a handle on webpack's configuration system adding these features is as easy as using the right plugins and loaders!
+2. Watch [this video](https://rapidevelop.org/webpack/setup-html-webpack-plugin) which demonstrates how to install `html-webpack-plugin` and set different options on it (note that some of the options shown are redundant as they are already the [default settings](https://github.com/jantimon/html-webpack-plugin#options)).
+    1. The [webpack website's tutorial](https://webpack.js.org/guides/output-management/) on output management also gives an example of using `html-webpack-plugin` but does not demonstrate using the template option. It also starts by demonstrating how to configure multiple entry points - you are unlikely to need this feature for the projects in this part of the course.
+3. Finally, the first part of [this tutorial](https://webpack.js.org/guides/development/) talks about source maps, a handy way to track down which source file (index.js, a.js, b.js) an error is coming from when you use webpack to bundle them together. This is essential to debugging bundled code in your browser's DevTools. If the error comes from b.js the error will reference that file instead of the bundle. It also walks through an example of the `--watch` feature and `webpack-dev-server` package, either of which will prove invaluable when working with webpack.
```

Comment:
Mentioning the source map here feels great!

Reply:
The source maps mention was already there but I did add the mentions of `webpack-dev-server` (and to ignore `webpack-dev-middleware` for Express.